cpp/ql/test/query-tests/Best Practices/Likely Errors/CommaBeforeMisleadingIndentation/test.cpp

```diff
@@ -9,11 +25,123 @@ void test(int i, int j, int (*foo)(int))
 		(void)i,	// BAD
 	(void)j;
 
-	foo((i++, j++)); // GOOD
-	foo((i++,        // GOOD
-	     j++));
-	foo((i++
-	   , j++));      // GOOD
-	foo((i++,
-	j++));           // BAD (?)
+	// Parenthesized comma (borderline example):
+
+	foo(i++), j++;   // GOOD
+	(foo(i++), j++); // GOOD
+	(foo(i++),       // GOOD
+	 j++);
+	(foo(i++),
+	j++);            // BAD (?)
+
+	x.foo(i++), j++;   // GOOD
+	(x.foo(i++), j++); // GOOD
+	(x.foo(i++),       // GOOD [FALSE POSITIVE]
+	 j++);
+	(x.foo(i++),
+	j++);              // BAD (?)
+
+	FOO(i++), j++;   // GOOD
+	(FOO(i++), j++); // GOOD
+	(FOO(i++),       // GOOD
+	 j++);
+	(FOO(i++),
+	j++);            // BAD (?)
+
+	(void)(i++), j++;   // GOOD
+	((void)(i++), j++); // GOOD
+	((void)(i++),       // GOOD [FALSE POSITIVE]
+	 j++);
+	((void)(i++),
+	j++);               // BAD (?)
+
+	// Comma in argument list doesn't count:
+
+	bar(i++, j++); // GOOD
+	bar(i++,
+	    j++);      // GOOD
+	bar(i++
+	  , j++);      // GOOD
+	bar(i++,
+	j++);          // GOOD: common pattern and unlikely to be misread.
+
+	BAR(i++, j++); // GOOD
+	BAR(i++,
+	    j++);      // GOOD
+	BAR(i++
+	  , j++);      // GOOD
+	BAR(i++,
+	j++);          // GOOD: common pattern and unlikely to be misread.
+
+	using T = decltype(x.foo(i++), // GOOD [FALSE POSITIVE]
+	                   j++);
+	(void)sizeof(x.foo(i++), // GOOD [FALSE POSITIVE]
+	             j++);
+
+	// Comma in loops
+
+    while (i = foo(j++), // GOOD
+           i != j && i != 42 &&
+               !foo(j)) {
+        i = j = i + j;
+    }
+
+    // Mixed tabs and spaces (ugly case):
+
+    for (i = 0,         // GOOD if tab >= 4 spaces else BAD
+		 j = 0;
+		 i + j < 10;
+         i++,           // GOOD if tab >= 4 spaces else BAD
+		 j++);
+
+	if (i)
+	    (void)i,	    // GOOD if tab >= 4 spaces else BAD
```

Comment:
Great tests / real world testing!

With a query like this I would be happy to simply not report dubious cases such as these three, and in general I wouldn't worry about missed results of any kind too much.

Reply:
Yes, but how would you distinguish between this case on line 97 and the one on [line 24](https://github.com/github/codeql/pull/10550/files/0e9b77e7c314509beea26479cb10b47a4044d5e0..cacf78838c61bb855e7bf212a77606552b2fe7a7#diff-5c2f6639161ed558bb0ba560dc0224ec85b97a24b1b17ed1f31af90b9d81c6e4R24-R26)? Without access to source code text, we cannot filter out the "dubious" cases (mixed tabs and spaces). So we have to alert for both cases, or neither.